### PR TITLE
add TaskDetailView

### LINKED ISSUE #137: TaskDetail view
It should be possible to pass a task to this component. This component will be used in the future either to create or to update a task.

```[tasklist]
### TasksDetail view
- [ ] #135 
- [ ] #136 
```

Group 45

### LINKED ISSUE #219: Change KanbanBoard Statemanagement
The state management of the kanbanboard doesn't work properly or isn't easy to integrate therefore we should change it.

Open because of PR #218

### PR DESCRIPTION
**Pull request by DasProffi**
closes #137
closes #219

### CODE REVIEW

**Comment by DasProffi** [comment]
Due to a bug with the kandban board state management, you need to update the task, update the patient and then switch away and back to see the updated kandban board.

This problem should be resolved in its own issue

File: lib/icons/Close.tsx
Comment (use-to) [inline lib/icons/Close.tsx:1]: Can't we use lucide.dev?

File: tasks/components/layout/TaskDetailView.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        <ToggleableInput
+          id={task.id}
+          value={task.name}
+          onChange={name => onChange({ ...task, name })}
+          labelClassName={tw('text-xl font-semibold')}
+        />
+        <div className={tw('flex flex-row')} onClick={onClose}>
+          <Close/>
+          <span className={tw('ml-2 hover:text-negative-500')}>{translation.close}</span>
+        </div>
```
Comment (use-to) [inline tasks/components/layout/TaskDetailView.tsx:86]: Change order to:
1. text
2. icon

File: tasks/components/layout/TaskDetailView.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  language,
+  task,
+  onChange,
+  onClose,
+  onFinishClick
+}: PropsWithLanguage<TaskDetailViewTranslation, TaskDetailViewProps>) => {
+  const translation = useTranslation(language, defaultTaskDetailViewTranslation)
+  return (
+    <div className={tw('flex flex-col h-full')}>
+      <div className={tw('flex flex-row justify-between items-center')}>
```
Comment (use-to) [inline tasks/components/layout/TaskDetailView.tsx:76]: apply more padding

File: tasks/components/layout/TaskDetailView.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        <ToggleableInput
+          id={task.id}
+          value={task.name}
+          onChange={name => onChange({ ...task, name })}
+          labelClassName={tw('text-xl font-semibold')}
+        />
```
Comment (use-to) [inline tasks/components/layout/TaskDetailView.tsx:82]: Toggle Input should be in edit state by default
Also the edit icon is way too far away from the actual input

**Review by use-to — CHANGES_REQUESTED** [review]

File: tasks/components/layout/PatientDetails.tsx [hunk 4]
```diff
@@ -54,14 +56,38 @@ export const PatientDetail = ({
 }: PropsWithLanguage<PatientDetailTranslation, PatientDetailProps>) => {
   const translation = useTranslation(language, defaultPatientDetailTranslations)
   const [newPatient, setNewPatient] = useState<PatientDTO>(patient)
+  const [newTask, setNewTask] = useState<TaskDTO|undefined>(undefined)
 
   return (
     <div className={tw('flex flex-col py-4 px-6')}>
+      <Modal
```
Comment (use-to) [inline tasks/components/layout/PatientDetails.tsx:63]: On overlay click = close modal

It feels just more naturally...

File: lib/icons/Close.tsx
Comment (DasProffi) [reply lib/icons/Close.tsx:1]: yes, it wasn't available as a package when I started writing on this braach

File: tasks/components/SubtaskView.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                    newSubtasks[index].name = text
+                    onChange(newSubtasks)
+                  }}
+                />
+                <button
+                  className={tw('text-hw-negative-400 hover:text-hw-negative-500 ml-4')}
+                  onClick={() => onChange(subtasks.filter((_, subtaskIndex) => subtaskIndex !== index))}
+                >
+                  {translation.remove}
+                </button>
```
Comment (use-to) [inline tasks/components/SubtaskView.tsx:77]: Can we take this out into another component?
- [ ] The checkbox should use proper colors
- [ ] What about using the _ToggableInput_ for subtasks here?
- [ ] If you're using _Toggablenput_ we can hide the deletion button within the edit mode.
- [ ] Use a delete icon instead

File: tasks/components/layout/TaskDetailView.tsx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+        <ToggleableInput
+          autoFocus
+          initialState="editing"
+          id={task.id}
+          value={task.name}
+          onChange={name => onChange({ ...task, name })}
+          labelClassName={tw('text-xl font-semibold')}
+        />
+        <button className={tw('flex flex-row hover:text-hw-negative-500 gap-x-2')} onClick={onClose}>
+          <span className={tw('text-black')}>{translation.close}</span>
```
Comment (use-to) [inline tasks/components/layout/TaskDetailView.tsx:86]: On hover: The text is still black not red (negative) is intended.

File: tasks/components/layout/TaskDetailView.tsx [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+                ]}
+                onChange={isPublicVisible => onChange({ ...task, isPublicVisible })}
+              />
+            </div>
+          </div>
+          {task.creationDate !== undefined ? (
+            <div className={tw('flex flex-col gap-y-8 mt-16')}>
+              <div className={tw('flex flex-col')}>
+                <span className={tw('font-bold text-gray-800 mb-1')}>{translation.creationTime}</span>
+                <span>{task.creationDate.toString()}</span>
```
Comment (use-to) [inline tasks/components/layout/TaskDetailView.tsx:150]: Use format string for better representation.
You're able to use the datetime provided by the os.

Empathize with a potential user... Is it relevant to see the timezone information? The exact second?

**Review by use-to — CHANGES_REQUESTED** [review]

File: tasks/components/layout/TaskDetailView.tsx (hunk 6 above)
Comment (DasProffi) [reply tasks/components/layout/TaskDetailView.tsx:86]: Red text did't look good

Comment (use-to) [reply tasks/components/layout/TaskDetailView.tsx:86]: So then remove this hover effect on that icon, too.
Think icon + text as an IconTextButton component. It should be consistent within itself.

**Comment by DasProffi** [comment]
> > Due to a bug with the kandban board state management, you need to update the task, update the patient and then switch away and back to see the updated kandban board.
> > This problem should be resolved in its own issue #219
> 
> Related to #219: The progress indicator on a regular task card is changing while editing the task. That's fine so far, but after adding new subtasks the progress will be calculated on the fly using the old amount of subtasks.

What do you mean by calculated on the fly, or what is your expected behavior of that component?

**Comment by DasProffi** [comment]
Toggleinput isn't working properly on the SubtaskTile, but this is to much for this PR we should fix that in a seperate issue

**Comment by use-to** [comment]
@DasProffi I think we have to meet for this specific bug. It's not relevant for now we can create a separate issue for that.

File: tasks/components/layout/TaskDetailView.tsx [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+  return (
+    <div className={tw('flex flex-col h-full p-2')}>
+      <div className={tw('flex flex-row justify-between items-center')}>
+        <ToggleableInput
+          autoFocus
+          initialState="editing"
+          id={task.id}
+          value={task.name}
+          onChange={name => onChange({ ...task, name })}
+          labelClassName={tw('text-xl font-semibold')}
```
Comment (use-to) [inline tasks/components/layout/TaskDetailView.tsx:83]: Can you increase the size a little bit?

**Review by use-to — CHANGES_REQUESTED** [review]

File: tasks/components/layout/TaskDetailView.tsx [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+  language,
+  task,
+  onChange,
+  onClose,
+  onFinishClick
+}: PropsWithLanguage<TaskDetailViewTranslation, TaskDetailViewProps>) => {
+  const translation = useTranslation(language, defaultTaskDetailViewTranslation)
+  return (
+    <div className={tw('flex flex-col h-full p-2')}>
+      <div className={tw('flex flex-row justify-between items-center')}>
```
[the remark this answers is not in the document]
Comment (use-to) [reply tasks/components/layout/TaskDetailView.tsx:76]: Maybe we can also use the vertical divider.
Similar to the TwoColumnComponent...

**Review by use-to — APPROVED** [review]
Really cool! 🚀

**Pull request merged**